CorsixTH/Lua/cheats.lua

```diff
@@ -60,11 +103,52 @@ function Cheats:performCheat(num)
   return cheat_success and self.cheat_list[num].name ~= "lose_level"
 end
 
+--! Checks the obfuscated cheat code for a match and executes it
+--!param num (number) The obfuscated cheat value
+--!return Returns name of the cheat executed from the lookup table, or nil
+function Cheats:processCheatCode(num)
+  for name, data in pairs(self.fax_cheats) do
+    if data.lower < num and data.upper > num then
+      self:toggleCheat(name)
+      return name
+    end
+  end
+  return -- cheat not found
+end
+
+
+--! Performs a cheat from fax_cheats
+--!param name (string) The cheat called from the list
+function Cheats:toggleCheat(name)
+  local ui = self.hospital.world.ui
+  local cheat = self.fax_cheats[name]
+  local cheatWindow = ui:getWindow(UICheats)
+  local speech
+  if not self.hospital.active_cheats[name] then
+    cheat.enable(self)
+    speech = self._getAdviserMessage(name, true)
+    self:announceCheat(speech)
+  else
+    cheat.disable(self)
+    speech = self._getAdviserMessage(name, false)
+    self:announceCheat(speech)
+  end
```

Comment:
Move repeated function call to 136
```suggestion
  else
    cheat.disable(self)
    speech = self._getAdviserMessage(name, false)
  end
  self:announceCheat(speech)
```